lib/checkfunctions.cpp

```diff
@@ -232,6 +232,12 @@ void CheckFunctions::checkMathFunctions()
                         (MathLib::isFloat(number) && MathLib::toDoubleNumber(number) <= 0.))
                         mathfunctionCallWarning(tok);
                 }
+                else if (Token::Match(tok, "log1p|log1pf|log1pl ( %num% )")) {
+                    const std::string& number = tok->strAt(2);
+                    if ((MathLib::isInt(number) && MathLib::toLongNumber(number) <= -1) ||
+                        (MathLib::isFloat(number) && MathLib::toDoubleNumber(number) <= -1.))
+                        mathfunctionCallWarning(tok);
+                }
```

Comment:
The only difference between this check and the one in rest of the log-functions is the constant (1 and 1.). It could be extracted to a function. Since it needs both an integer and a float, I couldn't really decided if I should pass a float and  cast to int (or vice versa) or pass both an int and a float (I didn't like any of them), or just pass a bool for log1p mode (but I liked that option even less). Thoughts?